oauthutil/oauthutil.go

```diff
@@ -438,6 +439,7 @@ func (s *authServer) Start() {
 		fs.Debugf(nil, "No code found on request")
 		w.WriteHeader(500)
 		fmt.Fprintf(w, "<h1>Failed!</h1>\nNo code found returned by remote server.")
+		s.code <- "-1"
```

Comment:
I think I'd probably change this to always return the code - something like this

```patch
--- a/oauthutil/oauthutil.go
+++ b/oauthutil/oauthutil.go
@@ -428,17 +428,19 @@ func (s *authServer) Start() {
 				fs.Debugf(nil, "Successfully got code")
 				if s.code != nil {
 					fmt.Fprintf(w, "<h1>Success</h1>\n<p>Go back to rclone to continue</p>")
-					s.code <- code
 				} else {
 					fmt.Fprintf(w, "<h1>Success</h1>\n<p>Cut and paste this code into rclone: <code>%s</code></p>", code)
 				}
 			}
 			return
+		} else {
+			fs.Debugf(nil, "No code found on request")
+			w.WriteHeader(500)
+			fmt.Fprintf(w, "<h1>Failed!</h1>\nNo code found returned by remote server.")
+		}
+		if s.code != nil {
+			s.code <- code
 		}
-		fs.Debugf(nil, "No code found on request")
-		w.WriteHeader(500)
-		fmt.Fprintf(w, "<h1>Failed!</h1>\nNo code found returned by remote server.")
-
 	})
 
 	var err error
```

What do you think?

Reply:
```
--- a/oauthutil/oauthutil.go
+++ b/oauthutil/oauthutil.go
@@ -428,17 +428,19 @@ func (s *authServer) Start() {
 				fs.Debugf(nil, "Successfully got code")
 				if s.code != nil {
 					fmt.Fprintf(w, "<h1>Success</h1>\n<p>Go back to rclone to continue</p>")
-					s.code <- code
 				} else {
 					fmt.Fprintf(w, "<h1>Success</h1>\n<p>Cut and paste this code into rclone: <code>%s</code></p>", code)
 				}
 			}
 -			return
+		} else {
+			fs.Debugf(nil, "No code found on request")
+			w.WriteHeader(500)
+			fmt.Fprintf(w, "<h1>Failed!</h1>\nNo code found returned by remote server.")
+		}
+		if s.code != nil {
+			s.code <- code
 		}
-		fs.Debugf(nil, "No code found on request")
-		w.WriteHeader(500)
-		fmt.Fprintf(w, "<h1>Failed!</h1>\nNo code found returned by remote server.")
-
 	})
 
 	var err error
```

By removing the return, it works

Reply:
looks good!